packages/api/internal/handlers/sandbox_logs.go

```diff
@@ -0,0 +1,94 @@
+package handlers
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"github.com/grafana/loki/pkg/loghttp"
+	"github.com/grafana/loki/pkg/logproto"
+	"go.opentelemetry.io/otel/attribute"
+
+	"github.com/e2b-dev/infra/packages/api/internal/api"
+	"github.com/e2b-dev/infra/packages/api/internal/constants"
+	"github.com/e2b-dev/infra/packages/api/internal/utils"
+
+	"github.com/e2b-dev/infra/packages/shared/pkg/models"
+	"github.com/e2b-dev/infra/packages/shared/pkg/telemetry"
+)
+
+const (
+	oldestLogsLimit = 168 * time.Hour // 7 days
+)
+
+func (a *APIStore) GetSandboxesSandboxIDLogs(
+	c *gin.Context,
+	sandboxID string,
+	params api.GetSandboxesSandboxIDLogsParams,
+) {
+	ctx := c.Request.Context()
+	sandboxID = utils.ShortID(sandboxID)
+
+	teamID := c.Value(constants.TeamContextKey).(models.Team).ID
+
+	telemetry.SetAttributes(ctx,
+		attribute.String("instance.id", sandboxID),
+		attribute.String("team.id", teamID.String()),
+	)
+
+	var start time.Time
+
+	end := time.Now()
+
+	if params.Start != nil {
+		start = time.UnixMilli(int64(*params.Start))
+	} else {
+		start = end.Add(-oldestLogsLimit)
+	}
+
+	fmt.Printf("Start: %v\n End: %v\n", start, end)
+
+	// Sanitize ID
+	// https://grafana.com/blog/2021/01/05/how-to-escape-special-characters-with-lokis-logql/
+	id := strings.ReplaceAll(sandboxID, "`", "")
```

Comment:
is there a possibility for some "loki injections"? We can make sure there are only letters and numbers and then we don't even need backticks

Reply:
Backticks are used as string literals in LogQL to escape everything. I could have escaped the backtick to prevent injection but I just removed it here.